src/HostManager.cpp

```diff
@@ -178,3 +163,34 @@ Host* HostManager::getHost(const QString& hostname)
 
     return pHost;
 }
+
+HostManager::Iter::Iter(HostManager* mgr, bool top)
+{
+    if (top) {
+        it = mgr->mHostPool.begin();
+    } else {
+        it = mgr->mHostPool.end();
+    }
+}
+
+bool HostManager::Iter::operator== (const Iter& other)
+{
+    return it == other.it;
+}
+
+bool HostManager::Iter::operator!= (const Iter& other)
+{
+    return it != other.it;
+}
+
+HostManager::Iter& HostManager::Iter::operator++()
+{
+    it++;
+    return *this;
+}
+
+QSharedPointer<Host> HostManager::Iter::operator*()
```

Comment:
Yeah, me too, but I couldn't figure out how to do that without running into a heap of random error messages. (I freely admit that my current C++ experience is seriously limited.)

IMHO the important bit is that the caller gets to use a straight `auto host : mHostManager`, but I'll happily clean it up any way you'd like if you point me to some code that looks vaguely like you'd like this to look. ;-)

Reply:
Ah I just meant the unique_ptr

Reply:
Yes I could, but that's intrusive because if you have `std::shared_ptr<X> a`and `X* b`,  comparing `a` and `b` for equality will no longer work(!!).

```
Host.cpp:507:23: error: no match for 'operator==' (operand types are 'std::shared_ptr<Host>' and 'Host*')
  507 |         if (otherHost == this || !otherHost->mpConsole) {
      |             ~~~~~~~~~ ^~ ~~~~
      |             |            |
      |             |            Host*
      |             std::shared_ptr<Host>
```
`QSharedPointer` doesn't have that problem. IMHO that's a glaring omission in the C++ stdlib, but they probably think that if you export something as a `<shared_ptr>`everybody and their dog should be forced to either use it that way or explicitly add a typecast.

So yes, replacing that thing is a good idea esp when you look at how many objects have a `Host* mpHost` member, but that's a different cleanup patch.